[2024/07/23] feature/marker/cu, rd >> Marker 기능 구현

LeeNaYoung240 opened the pull request:
## #️⃣ 연관된 이슈

> #8, #9, #10, #11, #17 
> close #8 
> close #9
> close #10
> close #11

## 📝 작업 내용

- Marker 생성
- Marker 글 작성
- Marker 글 수정
- Marker 전체 조회
- Marker 단건 조회
- Marker 삭제
- Marker Http 추가
- gitignore에 코드 추가


## 💬 리뷰 요구사항(선택)

> dailyPlan, Place 부분 구현이 되어있지 않아 연관된 부분 주석처리 후 TODO로 작성했습니다.
> Service 함수명 보시고 적절하지 않은 부분 알려주세요.

eunsaemsaem committed:
Merge from feat/marker

eunsaemsaem committed:
[#9] ✨ feat : 마커 생성 (dailyPlan, place 서비스 부분 수정 필요)

eunsaemsaem committed:
[#9] ✨ feat : 마커 관련 에러코드 추가

eunsaemsaem committed:
[#9] ✨ fix : 마커 생성 시 응답 메세지 수정

eunsaemsaem committed:
[#9] ✨ feat : 마커 글 생성 기능 추가

LeeNaYoung240 committed:
[#8] ✨ feat : 마커 전체 조회(dailyPlan, place 서비스 수정 필요)

- 마커 전체 조회 기능
- dailyPlan service 수정 필요

eunsaemsaem committed:
Merge branch 'feature/marker/rd' into feature/marker/cu

eunsaemsaem committed:
[#9] ✨ feat : 마커 글 생성 관련 공통 메서드 추가

- 동일한 유저인지 비교하는 메서드 추가
- 조회 관련 코드와 종합하여 정리
- 현재 사용할 수 없는 서비스 주석처리

LeeNaYoung240 committed:
[#8] 🩹 fix : 파일 충돌로 gitignore 코드 변경

LeeNaYoung240 committed:
[#8] ✨feat, 🩹 fix : 마커 단건 조회 기능 추가, 특정 데일리 플랜 마커 전체 조회 ResponseDto로 변경

- 마커 단건 조회 기능 추가
- ErrorCode 이름 변경
- 특정 데일리 플랜 마커 전체 조회 : Maker-> MarkerResponseDto 변경

LeeNaYoung240 committed:
[#8] 🩹 fix : 파일 충돌로 gitignore 코드 변경

LeeNaYoung240 committed:
[#8] 🩹 fix : 마커 기준 오름차순 정렬로 변경, Service 함수 밑으로 순서 변경

- 마커 오름차순 정렬로 코드 변경
- Service 부분 함수 밑으로 순서 변경

eunsaemsaem committed:
[#9] ✨ feat : 마커 글 수정 api 추가

- 마커 글 생성과 동일한 로직이므로 controller만 추가
- service 내 메서드 명을 updateMarkerContent로 변경

eunsaemsaem committed:
[#9] ✨ feat : 유저의 권한 확인 메서드 추가

- 유저 권한 관련 에러 코드 추가
- service의 모든 메서드에서 유저의 권한 확인하는 로직 추가

LeeNaYoung240 committed:
[#11] ✨ feat : MarkerStatusEnum 추가, Marker 삭제 기능 추가

- MarkerStatusEnum 추가
- Marker 삭제 기능 추가

LeeNaYoung240 committed:
[#11] ♻️ refactor : 중복된 코드 함수로 통합

- 중복된 코드 함수로 통합
- 마커 생성 반환값 수정
- 조회 url "s" 추가

LeeNaYoung240 committed:
[#11] ♻️ refactor : 중복된 코드 함수로 통합

Co-authored-by: eunsaemsaem <sweeet0308@gmail.com>

LeeNaYoung240 committed:
Merge branch 'dev' into feature/marker/rd

LeeNaYoung240 committed:
[#17] 🎉 add, ✨ feat : 마커 http 추가

Co-authored-by: eunsaemsaem <sweeet0308@gmail.com>

LeeNaYoung240 committed:
[#9] 🔥 del : 파일 삭제

LeeChangHyeong commented on http/http-client.private.env.json:
```diff
@@ -0,0 +1,5 @@
+{
+  "development": {
+    "ADMIN_TOKEN": "qlvmfpsemdpsmswjdakfajtwlrhrnldudns5auddmlroqkfwkrkdlTek"
```
P3 - 이 토큰은 숨길 수 있으면 숨겨서 올리는게 조금 더 좋을 것 같습니다! 숨길 수 있다면 수정 부탁드립니다~

LeeNaYoung240 replied:
삭제해서 수정했습니다!

LeeChangHyeong commented on src/main/java/com/befriend/detour/domain/marker/dto/MarkerContentRequestDto.java:
```diff
@@ -0,0 +1,10 @@
+
+public class MarkerContentRequestDto {
+
+    @Getter
```
Q - Getter를 class위에 달지 않고 필드에 단 이유가 있을까요?

LeeNaYoung240 replied:
잘못 달아서 class 위로 수정했습니다! 감사합니다

LeeChangHyeong commented on src/main/java/com/befriend/detour/domain/marker/entity/MarkerStatusEnum.java:
```diff
@@ -0,0 +1,20 @@
+        public static final String ACTIVE = "ACTIVE";
+        public static final String DELETED = "DELETED";
+    }
+
```
P3 - 공백 한칸 삭제해주시면 컨벤션에 더 적합할 것 같네요!

LeeNaYoung240 replied:
컨벤션에 맞게 수정했습니다!

LeeChangHyeong commented on src/main/java/com/befriend/detour/domain/marker/repository/MarkerRepository.java:
```diff
@@ -0,0 +1,10 @@
+
+@Repository
+public interface MarkerRepository extends JpaRepository<Marker, Long>, MarkerRepositoryCustom  {
+
```
P3- 여기도 공백 있습니다!

LeeNaYoung240 replied:
컨벤션에 맞게 수정했습니다!

LeeChangHyeong commented on src/main/java/com/befriend/detour/domain/marker/repository/MarkerRepositoryCustom.java:
```diff
@@ -0,0 +1,14 @@
+import java.util.Optional;
+
+public interface MarkerRepositoryCustom {
+    List<MarkerResponseDto> findByDailyPlanId(Long dailyPlanId);
```
P3 - 여기 위에도 한 칸 띄우시는게 컨벤션에 맞을 것 같아요

LeeNaYoung240 replied:
컨벤션에 맞게 수정했습니다!

LeeNaYoung240 committed:
[#9] 🔥 del : 파일 삭제

LeeChangHyeong commented on src/main/java/com/befriend/detour/domain/marker/repository/MarkerRepositoryImpl.java:
```diff
@@ -0,0 +1,41 @@
+
+    @Override
+    public List<MarkerResponseDto> findByDailyPlanId(Long dailyPlanId) {
+        return jpaQueryFactory.selectFrom(marker)
```
P3 - 여기 위에 띄우면 컨벤션에 더 맞을 것 같습니다~

LeeNaYoung240 replied:
컨벤션에 맞게 수정했습니다!

sihyun615 commented on src/main/java/com/befriend/detour/domain/marker/service/MarkerService.java:
```diff
@@ -0,0 +1,146 @@
+        - 지도에서 장소를 검색하고 선택한 후에 저장버튼을 누르면 마커 생성
+        - 프론트에서 placeId를 넘겨주면 해당 아이디로 place와 연관관계 설정
+         */
+        validateActiveUser(user);
```
P2 - 차단된 사용자는 이렇게 메서드마다 추가하기보다 로그인하는 부분에서부터 막는게 좋지 않을까요?

LeeNaYoung240 replied:
네! 맞습니다. 로그인 하는 부분에서 막도록 수정했습니다!

LeeChangHyeong commented on src/main/java/com/befriend/detour/domain/marker/repository/MarkerRepositoryImpl.java:
```diff
@@ -0,0 +1,41 @@
+        Marker result = jpaQueryFactory.selectFrom(marker)
+                .where(marker.id.eq(markerId)
+                        .and(marker.dailyPlan.id.eq(dailyPlanId)))
+                .fetchOne();
```
P3 - return 앞에 한칸 띄워주세요~

LeeNaYoung240 replied:
감사합니다! 컨벤션에 맞게 수정했습니다!

LeeNaYoung240 committed:
[#9] 🔥 del : 파일 삭제(gitignore)

LeeChangHyeong reviewed (commented):
코드 확인했습니다!
컨밴션 조금만 더 신경써 주세요!!

sihyun615 commented on src/main/java/com/befriend/detour/domain/marker/controller/MarkerController.java:
```diff
@@ -0,0 +1,81 @@
+    private final MarkerService markerService;
+
+    // 마커 생성
+    @PostMapping("/{dailPlanId}/markers/{placeId}")
```
P1 - dailyPlanId인데 오타난 것 같습니다!

LeeNaYoung240 replied:
오타 수정했습니다! 감사합니다!

sihyun615 reviewed (changes requested):
확인완료했습니다! 수고 많으셨어요☺️

LeeNaYoung240 committed:
[#20] 🩹 fix : 마커 url 수정, 컨벤션에 맞게 수정, user로 차단 변경

- 마커 url 수정 : dailyPlans -> daily-plans
- 컨벤션에 맞게 수정 : 공백 제거 및 공백 추가
- marker에서 차단된 사용자 접근 제한 -> user에서 처리

Co-authored-by: eunsaemsaem <sweeet0308@gmail.com>

eunsaemsaem committed:
Merge from dev

eunsaemsaem committed:
[#20] ♻️ refactor : dailyplan과 관련된 주석 해제

- DailyPlanService에 데일리플랜 찾는 메서드 추가
- 마커 조회 부분에 데일리플랜이 존재하는지 확인하는 코드 추가
